if-ancestor-frame-url won't work with site isolation

elijahsawyers opened the pull request:
#### 5c27a31ec45e7b94c727016c7be15c42e5152996
<pre>
if-ancestor-frame-url won&apos;t work with site isolation
<a href="https://bugs.webkit.org/show_bug.cgi?id=295541">https://bugs.webkit.org/show_bug.cgi?id=295541</a>
<a href="https://rdar.apple.com/155234942">rdar://155234942</a>

Reviewed by Brian Weinstein and Timothy Hatcher.

Revert &quot;dNR: fix sub_frame resourceType allowAllRequests rules&quot;
This reverts commit 34e4b871322e872acaed32b6cec020679fac098a.

Revert &quot;Introduce a new WebKit Content Blocking condition to act on the current frame or ancestor frame URL&quot;
This reverts commit 0d65d2e39211f8530206ac0d251c468d47cb1eaa.

Canonical link: <a href="https://commits.webkit.org/297098@main">https://commits.webkit.org/297098@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cdeb801c6509daad794084c9cbfcd85fa606abc8

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110509 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30168 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20601 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116535 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/60776 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/112472 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30847 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38756 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/84039 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113457 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24623 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99505 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64480 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/23985 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/17644 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/60330 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/94004 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17704 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119325 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37549 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27868 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/93000 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/37923 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95776 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92823 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23656 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37825 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15567 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33522 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37444 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/42915 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/37106 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40446 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/38815 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

b-weinstein commented on Source/WebCore/contentextensions/ContentExtensionsBackend.cpp:
```diff
@@ -256,23 +245,11 @@ ContentRuleListResults ContentExtensionsBackend::processContentRuleListsForLoad(
-    Vector<URL> ancestorSubframeURLs;
 
-    if (RefPtr frame = initiatingDocumentLoader.frame()) {
+    if (auto* frame = initiatingDocumentLoader.frame()) {
```
This could stay, but since you're doing a pure revert, I think it's fine to revert this.

elijahsawyers committed:
if-ancestor-frame-url won't work with site isolation
https://bugs.webkit.org/show_bug.cgi?id=295541
rdar://155234942

Reviewed by Brian Weinstein and Timothy Hatcher.

Revert "dNR: fix sub_frame resourceType allowAllRequests rules"
This reverts commit 34e4b871322e872acaed32b6cec020679fac098a.

Revert "Introduce a new WebKit Content Blocking condition to act on the current frame or ancestor frame URL"
This reverts commit 0d65d2e39211f8530206ac0d251c468d47cb1eaa.

Canonical link: https://commits.webkit.org/297098@main
